src/features/cart/CartItemControl.js

```diff
@@ -0,0 +1,46 @@
+import React from "react";
+import {
+  productAdded,
+  productQuantityDecremented,
+  productDeleted,
+} from "./cartSlice";
+import { useDispatch } from "react-redux";
+
+export default function CartItemControl({ quantity, productId }) {
+  const dispatch = useDispatch();
+
+  function onIncrementQuantity(event) {
+    const clickedProductId = parseInt(event.target.value, 10);
+
+    dispatch(productAdded({ id: clickedProductId }));
```

Comment:
productAdded with only 'id' as payload is a bit dangerous. it works just in case the product is already present in cart. if not it will corrupt the cart, since it will insert a product with only id and quantity.
either create an additional reducer productQuantityIncremented which only expects id.
or catch the case of  payload = '{id: xx}' in the current productAdded reducer

Reply:
check